src/test/java/org/jilt/test/data/annotations/PropagatedAnnotationName.java

```diff
@@ -0,0 +1,23 @@
+package org.jilt.test.data.annotations;
+
+import org.jilt.Builder;
+import org.jilt.BuilderStyle;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+@Builder(style = BuilderStyle.STAGED_PRESERVING_ORDER)
+public class PropagatedAnnotationName {
+  public final String firstName;
+  public final String middleName;
+
+  @Nonnull
+  @CheckForNull
+  public final String lastName;
```

Comment:
Also annotate this field with an annotation that cannot be placed on a parameter, to make sure the filtering works correctly.

If you can't find an annotation like this in the current project or its dependencies, feel free to define one next to this class.